src/main/java/net/minecraftforge/event/AddReloadListenerEvent.java

```diff
@@ -0,0 +1,34 @@
+package net.minecraftforge.event;
+
+import com.google.common.collect.ImmutableList;
+import net.minecraft.resources.DataPackRegistries;
+import net.minecraft.resources.IFutureReloadListener;
+import net.minecraftforge.common.MinecraftForge;
+import net.minecraftforge.eventbus.api.Event;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * The main ResourceManager is recreated on each reload, through {@link DataPackRegistries}'s creation.
+ *
+ * The event is fired on each reload and lets modders add their own ReloadListeners.
+ * The event is fired on the {@link MinecraftForge#EVENT_BUS}
+ */
+public class AddReloadListenerEvent extends Event
```

Comment:
Might want to emphasize this is for server reload listeners. Only clue in the javadoc is the link to data pack registries, but I could see someone getting confused and trying to use this for the client side resource pack reload listeners.